internal/controllers/clusterextension_controller.go

```diff
@@ -332,26 +331,20 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
 	}
 
 	for _, obj := range relObjects {
-		uMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
-		if err != nil {
-			setInstalledStatusConditionFailed(ext, fmt.Sprintf("%s:%v", ocv1alpha1.ReasonCreateDynamicWatchFailed, err))
-			return ctrl.Result{}, err
-		}
-
-		unstructuredObj := &unstructured.Unstructured{Object: uMap}
 		if err := func() error {
 			r.dynamicWatchMutex.Lock()
 			defer r.dynamicWatchMutex.Unlock()
 
-			_, isWatched := r.dynamicWatchGVKs[unstructuredObj.GroupVersionKind()]
+			_, isWatched := r.dynamicWatchGVKs[obj.GetObjectKind().GroupVersionKind()]
 			if !isWatched {
 				if err := r.controller.Watch(
-					source.Kind(r.cache, unstructuredObj),
-					crhandler.EnqueueRequestForOwner(r.Scheme(), r.RESTMapper(), ext, crhandler.OnlyControllerOwner()),
-					helmpredicate.DependentPredicateFuncs()); err != nil {
+					source.Kind(r.cache,
+						obj,
+						crhandler.EnqueueRequestForOwner(r.Scheme(), r.RESTMapper(), ext, crhandler.OnlyControllerOwner()),
+						helmpredicate.DependentPredicateFuncs())); err != nil {
 					return err
 				}
-				r.dynamicWatchGVKs[unstructuredObj.GroupVersionKind()] = struct{}{}
+				r.dynamicWatchGVKs[obj.GetObjectKind().GroupVersionKind()] = struct{}{}
```

Comment:
There's a different version of this in: #905 
https://github.com/operator-framework/operator-controller/pull/905/files#diff-a54f79b762054ccdc31e4011d28cce22af8f54d43e1fddc75745ce042f177f7e

Reply:
Could we use the unstructured one for now and have this PR merged? https://github.com/operator-framework/operator-controller/pull/905#discussion_r1628050929

Just curious if directly setting watches from typed objects would break anything.

Reply:
+1 @varshaprasad96. I don't know _for sure_ that using a structured type will break things, but I have at least some confidence that it might. So I think we should stick with the unstructured type for now.

Reply:
So, I tested this, with help from @joelanford, and it appears to work.